sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Compose.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.transforms;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+/**
+ * Function composition for Serializable {@code Comparator} objects and for
+ * {@code SerializableFunction} objects.
+ */
+public class Compose {
+  /**
+   * Compose a Serializable {@code Comparator} with a {@code SerializableFunction}.  The function
+   * is run on the input data, and the output of the function is run through the Comparator.
+   *
+   * @param comparator Comparator object
+   * @param fn         function to map raw inputs into a format understandable by the Comparator
+   * @param <T>        type of Comparator
+   * @param <CompT>    Comparator type
+   * @param <X>        raw input type
+   * @return Comparator
+   */
+  public static <T, CompT extends Comparator<T> & Serializable, X> Comp<X> of
```

Comment:
Java's `&` operator is not really first class - there are lots of important places you can't use it. So we have `SerializableComparator` already, and this method should work with that.